### PR TITLE
Fix `GenericRouteShield` styling.

### PR DESCRIPTION
**Pull request by MaximAlien**
Closing #3336.

### CODE REVIEW

File: Sources/MapboxNavigation/InstructionPresenter.swift [hunk 1]
```diff
@@ -179,7 +201,8 @@ class InstructionPresenter {
             attachment.image = image
         } else {
             let view = GenericRouteShield(pointSize: dataSource.font.pointSize, text: text)
-            view.foregroundColor = dataSource.textColor
+            view.foregroundColor = GenericRouteShield.appearance().foregroundColor
+            view.borderWidth = GenericRouteShield.appearance().borderWidth
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/InstructionPresenter.swift:205]: @avi-c, it's been a while since this logic was modified, but do you remember why you've decided to use `dataSource.textColor` instead of `GenericRouteShield`s appearance. Maybe I'm missing some edge case here.

File: Tests/MapboxNavigationTests/ManeuverViewSnapshotTests.swift [hunk 2]
```diff
@@ -60,18 +68,19 @@ class ManeuverViewSnapshotTests: TestCase {
...
+                            as: .image(precision: 0.95))
     }
 
     func testRightUTurn() {
-        maneuverView.drivingSide = .left
         maneuverView.visualInstruction = maneuverInstruction(.turn, .uTurn)
-        assertImageSnapshot(matching: maneuverView.layer, as: .image(precision: 0.95))
+        maneuverView.drivingSide = .left
+
+        assertImageSnapshot(matching: UIImageView(image: maneuverView.imageRepresentation),
```
Comment (MaximAlien) [inline Tests/MapboxNavigationTests/ManeuverViewSnapshotTests.swift:82]: Using `UIView.imageRepresentation` makes sure to verify whether flipping transformations were applied. This is relevant for right/left driving side roads.

File: Sources/MapboxNavigation/InstructionPresenter.swift (hunk 1 above)
Comment (1ec5) [reply Sources/MapboxNavigation/InstructionPresenter.swift:210]: #2280 was intended to fix #2161, but #2977 ended up actually fixing that issue when switching between day and night. Not sure if the changes in #2280 are still required.

File: Sources/MapboxNavigation/InstructionPresenter.swift [hunk 3]
```diff
@@ -179,7 +206,8 @@ class InstructionPresenter {
             attachment.image = image
         } else {
             let view = GenericRouteShield(pointSize: dataSource.font.pointSize, text: text)
```
Comment (1ec5) [inline Sources/MapboxNavigation/InstructionPresenter.swift:208]: Should the font also come from the appearance proxy?

File: Sources/MapboxNavigation/InstructionPresenter.swift (hunk 1 above)
Comment (1ec5) [reply Sources/MapboxNavigation/InstructionPresenter.swift:210]: One thing to watch out for is that CarPlay always uses the night style, even during the daytime when the phone uses the day style.

Comment (MaximAlien) [reply Sources/MapboxNavigation/InstructionPresenter.swift]: Behavior was changed: user will have to provide two different appearances for iOS and CarPlay, Navigation SDK by default also provides colors for different interface styles.

Comment (avi-c) [reply Sources/MapboxNavigation/InstructionPresenter.swift]: Your first comment got buried under my email notifications. 

I can't recall exactly why I chose the data source over the straight appearance proxy but it might have been that I was using the shield within another view container and needed it to defer to the datasource to find the right color.

File: Sources/MapboxNavigation/InstructionPresenter.swift (hunk 3 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/InstructionPresenter.swift:213]: Font is applied to `NSTextAttachment` afterwards, so I don't think it's relevant for cache. I'll run some manual tests though and consider adding unit-test.

File: Sources/MapboxNavigation/GenericRouteShield.swift [hunk 4]
```diff
@@ -94,25 +94,26 @@ public class GenericRouteShield: StylableView {
...
+            appearance.foregroundColor,
+            appearance.borderColor,
+            appearance.borderWidth,
+            appearance.cornerRadius,
+            traitCollection.userInterfaceIdiom.rawValue
+        ]
+        
+        if #available(iOS 12.0, *) {
+            criticalProperties.append(traitCollection.userInterfaceStyle.rawValue)
+        }
```
Comment (1ec5) [inline Sources/MapboxNavigation/GenericRouteShield.swift:115]: Thanks, this is a lot simpler.

File: Sources/MapboxNavigation/Style.swift [hunk 5]
```diff
@@ -49,6 +49,15 @@ open class Style: NSObject {
...
     @objc open func apply() { }
     
+    /**
+     Applies the style based on provided `UITraitCollection`.
+     
+     - parameter traitCollection: `UITraitCollection` instance, properties of which
+     (e.g. `userInterfaceStyle`, `userInterfaceIdiom` etc), will be used for applying style.
+     */
+    @available(iOS 12.0, *)
+    open func apply(for traitCollection: UITraitCollection?) { }
```
Comment (1ec5) [inline Sources/MapboxNavigation/Style.swift:59]: Let’s mention the addition of this method in the changelog.

**Review by 1ec5 — APPROVED** [review]

File: Sources/MapboxNavigation/Style.swift (hunk 5 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/Style.swift]: Decided not to add this public method. All styling configuration for CarPlay will happen in `DayStyle` (in case if CarPlay supports both light and dark styling).
